src/components/image_upload.tsx

```diff
@@ -30,71 +38,101 @@ export interface ImageUploadProps {
     imageURL?: string;
     /** Placeholder label */
     placeholderLabel?: string;
+    /**
+     * Whether cropping should be enabled. If enabled, the user is allowed to choose what part of the image to use.
+     * For every change, the `onFileRead` callback is called.
+     **/
+    allowCropping?: boolean;
 }
 
 export interface ImageUploadState {
+    /** Whether the user is currently dropping a file in this component **/
     dropActive: boolean;
+    /** The URL the user started cropping with. This is used to keep the whole image even then the imageURL changed **/
+    cropURL: string | null;
 }
 
+/**
+ * Component that allows a user to upload (and optionally crop) an image.
+ */
 export class ImageUpload extends React.Component<ImageUploadProps, ImageUploadState> {
 
     public static defaultProps: Partial<ImageUploadProps> = {
         shape: 'round',
-        size: '18rem'
+        size: '18rem',
     };
 
-    state = {
-        dropActive: false
-    }
+    state: ImageUploadState = {
+        dropActive: false,
+        cropURL: null,
+    };
 
     constructor(props) {
         super(props);
 
         this._onDropHandler = this._onDropHandler.bind(this);
         this._onDragEnter = this._onDragEnter.bind(this);
         this._onDragLeave = this._onDragLeave.bind(this);
+        this._onCropCancel = this._onCropCancel.bind(this);
     }
 
-    _onDropHandler(files: ImageFile[]): void {
-        this.setState({
-            dropActive: false
-        });
-
+    private _onDropHandler(files: ImageFile[]): void {
         const file = files[0];
+        this.setState({ dropActive: false });
+
+        const {allowCropping, onFileSelection, onFileRead} = this.props;
 
-        const {onFileRead, onFileSelection} = this.props;
         if (onFileSelection) {
             onFileSelection(file);
         }
+
         if (onFileRead) {
             const reader = new FileReader();
-            reader.onload = () => onFileRead(file, reader.result as string);
-            reader.onerror = console.error; // TODO
+            reader.onload = () => onFileRead(reader.result as string);
+            reader.onerror = console.error;
             reader.readAsDataURL(file);
         }
+
+        if (allowCropping) {
+            return this.setState({ cropURL: file.preview });
```

Comment:
This method has a return type `void`, so either that type or this line should be changed.